### PR TITLE
create-resources-at-scale-scripts

### PR DESCRIPTION
**Pull request by jerry-li-dev**
## Checklist

- [ ] Scripts in this pull request are written for the `bash` shell.
- [ ] This pull request was tested on __at least one of__ the following platforms:
  - [ ] Linux
  - [ ] Azure Cloud Shell
  - [ ] macOS
  - [ ] Windows Subsystem for Linux
- [ ] The most recent test date and test method are recorded in the script file.
- [ ] Scripts do not contain passwords or other secret tokens that are not randomized.
- [ ] No deprecated commands or arguments are used. ([Release notes](https://docs.microsoft.com/cli/azure/release-notes-azure-cli))
- [ ] All Azure resource identifiers which must be universally unique are guaranteed to be so.
- [ ] Resource names use a random function to ensure scripts can be run multiple times in quick succession without error.
- [ ] All scripts can be run in their entirely without user input.

### Testing information

CLI version:
```
az --version
```

Extensions required:

### CODE REVIEW

File: create-resources-at-scale/create-resources-at-scale.sh [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#!/bin/bash
+
+# Passed validation in Azure Cloud Shell Bash environment on 4/25/2022
+
+# Variable block
+# These variables have placeholder values that are replaced with values from the csv input file.
+Identifier="0"
```
Comment (dbradish-microsoft) [inline create-resources-at-scale/create-resources-at-scale.sh:7]: This is your CSV line number, correct? 

When creating a resource group, you'll need

```
let "randomIdentifier=$RANDOM*$RANDOM"
location="eastus"
resourceGroup="msdocs-script-rg-$randomIdentifier"
az group create --name $resourceGroup --location $location
```

File: create-resources-at-scale/create-resources-at-scale.sh [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+# task 1 create resources
+for i in $(seq 0 1);
+do
+    # Read in commands from csv file
+    echo read in commands from csv file
+    c1= sed -n "$Identifier"p file.csv
+    Identifier=$((Identifier+1))
+    echo $Identifier. Creating resources for $user
+    resourceGroup="ubuntu-vm-group22-$Identifier"
```
Comment (dbradish-microsoft) [inline create-resources-at-scale/create-resources-at-scale.sh:27]:
```
 resourceGroup="msdocs-script-rg-$randomIdentifier"
```

File: create-resources-at-scale/create-resources-at-scale.sh [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+createvm="true"
+createvnet="true"
+createrg="true"
+subnet=""
+vmName="msdocs-linuxVM-$user"
+vnetname=""
+vmImage="Ubuntu2204"
+location=""
+resourceGroup="msdocs-ubuntu-vm-group22-$Identifier"
+
```
Comment (dbradish-microsoft) [inline create-resources-at-scale/create-resources-at-scale.sh:18]: setupFileLocation="C:\myPath\myFileName"

File: create-resources-at-scale/create-resources-at-scale.sh [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    az group create --l $location -n $resourceGroup
+    echo $Identifier
+    echo test echo
+    if "$createvm" =true; then 
+        echo create vm
+        az vm create \
+            --resource-group $resourceGroup \
+            --name $vmName \
+            --image $vmImage \
+            --public-ip-sku Standard \
```
Comment (dbradish-microsoft) [inline create-resources-at-scale/create-resources-at-scale.sh:38]: Sku type should be coming from the CSV file and be a passed parameter

File: create-resources-at-scale/create-resources-at-scale.sh [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+            --admin-username $user\
+            --admin-password qwertY123456
```
Comment (dbradish-microsoft) [inline create-resources-at-scale/create-resources-at-scale.sh:40]: Password should be coming from a parameter value that you create with "msdocs-script-ps-$randomIdentifier". That password needs to be written to the TXT log file.

**Review by dbradish-microsoft — CHANGES_REQUESTED** [review]

File: create-resources-at-scale/create-resources-at-scale.sh [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+vmImage="Ubuntu2204"
+location=""
+resourceGroup="msdocs-ubuntu-vm-group22-$Identifier"
+
+# task 1 create resources
+for i in $(seq 0 1);
+do
+    # Read in commands from csv file
+    echo read in commands from csv file
+    c1= sed -n "$Identifier"p file.csv
```
Comment (dbradish-microsoft) [inline create-resources-at-scale/create-resources-at-scale.sh:24]: When reading from a CSV or JSON file, don't you need to use [jq](https://jqlang.github.io/jq/manual/)?

# Show the contents of a file in the console
fileName=msdocs-tutorial.json
cat $fileName | jq

# Get a JSON dictionary object
stgKV=$(jq -r '.environments.stg."kv-secretName"' $fileName)
echo $stgKV

# Filter a JSON array
devKV=$(jq -r '.environments.dev[] | select(.status=="active") | ."kv-secretName"' $fileName)
echo $devKV

File: create-resources-at-scale/create-resources-at-scale.sh (hunk 5 above)
Comment (jerry-li-dev) [reply create-resources-at-scale/create-resources-at-scale.sh]: I have added this in with the latest PR.

File: create-resources-at-scale/create-resources-at-scale.sh (hunk 4 above)
Comment (jerry-li-dev) [reply create-resources-at-scale/create-resources-at-scale.sh:41]: I have added this in with the latest PR.

File: create-resources-at-scale/create-resources-at-scale.sh [hunk 7]
```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+# Passed validation in Azure Cloud Shell Bash environment on 4/25/2022
+
+# Variable block
+# These variables have placeholder values that are replaced with values from the csv input file.
+lineCounter="0"
```
Comment (dbradish-microsoft) [inline create-resources-at-scale/create-resources-at-scale.sh:7]: Add an **subscriptionName** parameter as every script needs this at the top -- there is no default subscription capability right now in azCLI.

File: create-resources-at-scale/create-resources-at-scale.sh [hunk 8]
```diff
@@ -0,0 +1,56 @@
...
+vmName="msdocs-linuxVM-$user"
+vnetname=""
+vmImage="Ubuntu2204"
+location=""
+resourceGroup="msdocs-ubuntu-vm-group22-$Identifier"
+setupFileLocation="C:\myPath\myFileName"
+publicIpSku=""
+
+# task 1 create resources
+for i in $(seq 0 1);
```
Comment (dbradish-microsoft) [inline create-resources-at-scale/create-resources-at-scale.sh:26]: between line 23 and 24, put in a `az account set --name $subscriptionName`

**Review by dbradish-microsoft — CHANGES_REQUESTED** [review]

File: create-resources-at-scale/create-resources-at-scale.sh (hunk 4 above)
Comment (dbradish-microsoft) [reply create-resources-at-scale/create-resources-at-scale.sh]: @jerry-li-dev , please click **Resolve conversation** when request for changes is complete. Thank you.

**Pull request closed without merging**